### PR TITLE
second_task_v2

### PR DESCRIPTION
**Pull request by kurkenov**

### CODE REVIEW

File: second_task.tf [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = "Main_Interenet_Gateway"
+  }
+}
+
+resource "aws_eip" "nat_eip" {
```
Comment (geoffrey1330) [inline second_task.tf:52]: Add 2 aws_eip and associate them each to the nat_gateway

File: second_task.tf [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  tags = {
+    Name = "MAIN_NAT2_GW"
+  }
+}
+
+
+
+
+resource "aws_route_table" "public2" {
```
Comment (geoffrey1330) [inline second_task.tf:142]: you should have only 1 public route table and 2 public route table association 
i.e delete second public route table and then associate the first route table to the second public route table association. 
```
resource "aws_route_table_association" "public2" {
  subnet_id      = aws_subnet.public2.id
  route_table_id = aws_route_table.public.id
}
```

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: second_task.tf (hunk 1 above)
Comment (kurkenov) [reply second_task.tf:52]: @geoffrey1330 
Have you checked ?

resource "aws_nat_gateway" "nat2" {
  allocation_id = aws_eip.nat_eip.id
  subnet_id     = aws_subnet.public2.id

  tags = {
    Name = "MAIN_NAT2_GW"
  }
}

130-137 lines

File: second_task.tf [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.igw.id
+  }
+
+  tags = {
+    Name = "PublicRouteTable2"
+  }
+}
+
```
Comment (geoffrey1330) [inline second_task.tf:164]: Add the second route table association. 
```
resource "aws_route_table_association" "public2" {
  subnet_id      = aws_subnet.public2.id
  route_table_id = aws_route_table.public.id
}
```

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: second_task.tf (hunk 3 above)
Comment (kurkenov) [reply second_task.tf:168]: https://github.com/kurkenov/devops-quiz1/pull/4/commits/4a8226f796d82e5249d6d2716581ca60713481a8
Done

File: second_task.tf [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  tags = {
+    Name = "MAIN_NAT2_GW"
+  }
+}
+
+
+
+
+resource "aws_route_table" "public2" {
```
Comment (geoffrey1330) [inline second_task.tf:156]: remove line 156 to 167 as they are not needed the 2 public route table association are making use of first route table

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: second_task.tf (hunk 4 above)
Comment (kurkenov) [reply second_task.tf]: Fixed https://github.com/kurkenov/devops-quiz1/pull/4/commits/18e658e297e46e82051a2a94a1fa1a545e30afd3

**Review by geoffrey1330 — APPROVED** [review]

**Pull request merged**